fuse/eval/metrics/sequence_gen/metrics_seq_gen_common.py

```diff
@@ -18,13 +18,99 @@
 """
 from typing import Optional, Tuple, List
 from functools import partial
-
+from copy import copy
 import torch
 import numpy as np
 
 from fuse.eval.metrics.metrics_common import MetricPerBatchDefault
 
 
+class MetricCountSeqAndTokens(MetricPerBatchDefault):
+    """
+    Counts the total number sequences and tokens in encoder_input
+    """
+
+    def __init__(
+        self,
+        encoder_input: str,
+        ignore_index: Optional[int] = None,
+        state: Optional[dict] = None,
+        **kwargs: dict,
+    ) -> None:
+        """
+        :param encoder_input: key to the encoder_input
+        :param ignore_index: token_id to ignore (not to count), typically pad token id
+        :param state: the sequence count and token count to continue for. Should be restored when we continue training.
+                    use get_state() to get the state and save it upon checkpointing,
+        :param kwargs: additional super class arguments
+        """
+        super().__init__(
+            seq_num="seq_num",  # collect log_probs - output of  _count_seq_and_tokens_update
```

Comment:
obsolete comments in this line and the following one

Reply:
👍